rosbridge_server/scripts/rosbridge_websocket.py

```diff
@@ -332,31 +326,48 @@ def protocol_parameter_handling(self):
         CallService.services_glob = RosbridgeWebSocket.services_glob
 
 
-def main(args=None):
-    if args is None:
-        args = sys.argv
+def start_ros_thread(node, shutdown_event):
+    def spin_ros():
+        executor = rclpy.executors.SingleThreadedExecutor()
+        executor.add_node(node)
+        while rclpy.ok() and not shutdown_event.is_set():
+            executor.spin_once(timeout_sec=1.0)
```

Comment:
Worried about this timeout being a hard-coded value, although it is just one spin...

Reply:
We need **some** timeout to make sure the thread wakes up once in a while to check for shutdown event. Alternatively, I can try running `executor.spin()` and calling `executor.shutdown()` from the main thread.

Reply:
I tried that and now one of the tests is timing out. I'm changing this PR to draft until I'll figure this out

Reply:
The tests are timing out due to bugs in rclpy. I submitted a PR with the fix in ros2/rclpy#1469.